Added User Prompt Script

kowsisoundhar12 opened the pull request:
Test_TC_MC_1_1.yaml
Test_TC_MC_1_2.yaml
Test_TC_OO_1_1.yaml
Test_TC_PCC_2_2.yaml
Test_TC_PCC_2_3.yaml
Test_TC_PCC_2_4.yaml
Test_TC_RH_1_1.yaml
Test_TC_RH_2_1.yaml
Test_TC_TM_1_1.yaml
Test_TC_TSUIC_1_1.yaml
[Test-exe-logs-apr5.zip](https://github.com/project-chip/connectedhomeip/files/8418856/Test-exe-logs-apr5.zip)

woody-apple reviewed (approved):
Fast tracking new tests

woody-apple commented:
@kowsisoundhar12 Please note there are conflicts with mainline!

bzbarsky-apple commented on src/app/tests/suites/certification/Test_TC_PCC_2_3.yaml:
```diff
@@ -63,13 +63,20 @@ tests:
-      command: "readAttribute"
-      attribute: "EffectiveControlMode"
+    - label:
+          "Read EffectiveControlMode attribute from the DUT and Verify that the
+          DUT response indicates EffectiveControlMode attribute has the value 1"
```
Why is this being done via UserPrompt?  Why not just check the value directly?

bzbarsky-apple commented on src/app/tests/suites/certification/Test_TC_MC_1_2.yaml:
```diff
@@ -49,42 +49,68 @@ tests:
-          constraints:
-              type: map32
+    - label:
+          "Read FeatureMap attribute from the DUT and Verify that the DUT
+          response indicates either value 3 or throws a general error if the
+          attribute is not supported"
```
Why is this not an automated test?

bzbarsky-apple commented on src/app/tests/suites/certification/Test_TC_PCC_2_2.yaml:
```diff
@@ -35,13 +35,21 @@ tests:
+      verification: |
+          ./chip-tool pumpconfigurationandcontrol read effective-operation-mode 1 1
+      cluster: "LogCommands"
+      command: "UserPrompt"
```
Why is this UserPrompt?  This should be an attribute read.

bzbarsky-apple commented on src/app/tests/suites/certification/Test_TC_PCC_2_2.yaml:
```diff
@@ -50,13 +58,21 @@ tests:
+      verification: |
+          ./chip-tool pumpconfigurationandcontrol read effective-operation-mode 1 1
+      cluster: "LogCommands"
+      command: "UserPrompt"
```
Again, why is this UserPrompt?

What is the actual goal of this PR?  Why are we making these changes?

raju-apple commented:
@bzbarsky-apple The goal of this PR is as follows. Currently we have a few steps disabled across many test scripts . Each of these scripts currently when run with TH can't be used for certification since its not a complete TC run. So instead of folks manually running all these "partially automated" tests , we want to put up a user prompt for every disabled step in the scripts for now so that the manual effort in these tests is limited to just that particular step and the rest of the TC can be run with the script. As and when we have a way to script/automate it we will remove the user prompt then. Does that make sense ?

bzbarsky-apple commented:
@raju-apple That makes sense, thank you.  In that case some specific questions about the actual test changes being made here.

In `Test_TC_MC_1_1.yaml`:
1. Why are we disabling automated testing (in CI) of the things that _are_ testable automatically (e.g. the ClusterRevision attribute for the "Low Power" cluster) by removing the .yaml from automated testing entirely?  Why are we removing anything at all from CI testing here, in fact?
2. Why are we making the command list attributes manually tested while leaving the attribute list automatically tested?  It's really not clear to me what the distinction here is.
3. Why is the test for feature map on the "Low Power" cluster saying it needs to be 3 or error?  That cluster does not have a feature map attribute defined, so per spec (after the most recent updates) reading the feature map should return 0.  This is not correctly implemented in the SDK yet, but that's what the test _should_ be testing.

In `Test_TC_OO_1_1.yaml`:
4. Why are the verification steps listing things like `...Cluster: 0x0000_0006 Attribute 0x0000_FFF9DataVersion: 1805855192` (not lack of space before `DataVersion`) which don't match the actual tool output, at least on any sort of recent commit?

In `Test_TC_PCC_2_2.yaml`:
5. Why are we replacing a perfectly automatable test (reads an attribute, checks that its value is 1) with a UserPrompt test?  If the issue is that the test currently fails in CI, then we should either fix the example apps used in CI or add a PICS value we can use to disable tests in CI or something.  (Note: this specific test is failing because the pump configuration and control cluster is not really implemented, as far as I can tell....)

There are similar issues in a bunch of the other files (e.g. the relative humidity MaxMeasuredValue should not be a user prompt because we can just test it automatically, etc), but let's start with those.

kvikrambhat commented:
@bzbarsky-apple 
Test_TC_MC_1_1.yaml:
1.We had an issues on the CI where we saw the builds fail for Darwin which had user prompts we have filed and issue and had decided to make the initial PR with them removed from Darwin [https://github.com/project-chip/connectedhomeip/issues/17089]
2. The attributes that are currently supported by the YAML framework are automated, attributes that are currently missing from the YAML framework we have converted to prompts.
3.The verification steps are picked directly from the Test Event validation spread sheet, we are assuming it to be the single source of truth for validation steps as it was use during the test events.

Test_TC_OO_1_1.yaml:
4. same as #3 above, these are directly copied over from the spread sheet.

Test_TC_PCC_2_2.yaml
5. This is a tricky question, previously when the scripts where added we only had the option to disable test steps when we saw a failure.Support for PICS, Optional tags had not been added, now that we support these we might have to use these according to the issue at hand, we had decided to convert all disabled steps to prompts but now that i think of it it might be better to test the disabled steps and decide based on the type of issue.
   a. YAML Support Issue : If the YAML framework does not support an attribute we will need to use a user prompt
   b. Sample App Issue : If the same app is missing any attribute or feature we will need to use PICS or Optional Tags [Based on what is defined in the spec] 
 @raju-apple let me know what you think would be the best approach to reduce the number of disabled steps

bzbarsky-apple commented:
> we have filed and issue

Just now, yes?  Anyway, fix for that issue is up.  Next time, please start with the issue, not disabling tests...

> 2. The attributes that are currently supported by the YAML framework are automated, attributes that are currently missing from the YAML framework we have converted to prompts.

The command list attributes are supported by the YAML framework last I checked, so this answer doesn't seem to answer the question I asked.

> 3.The verification steps are picked directly from the Test Event validation spread sheet, we are assuming it to be the single source of truth for validation steps as it was use during the test events.

Is this the spreadsheet at https://groups.csa-iot.org/wg/matter-csg/document/25323?downloadRevision=49161 ?  That seems to match the test plan at  https://github.com/CHIP-Specifications/chip-test-plans/blob/c4546100319afa15fff4448d0cd9f1f772876374/src/cluster/media.adoc#tc-mc-1-1-global-attributes-low-power-cluster-dut-as-server (revision as of today), for TC-MC-1.1 step 2.  And does not match the YAML being added in this PR as far as I can tell.  Am I missing something?  Or is there a different spreadsheet involved?

> Test_TC_OO_1_1.yaml: 4. same as #3 above, these are directly copied over from the spread sheet.

Please raise an issue on the spreadsheet/test-plan on the claimed output here not matching the actual tool output?

> but now that i think of it it might be better to test the disabled steps and decide based on the type of issue

That would be much appreciated, yes.  Your proposed breakdown here makes sense to me.

kvikrambhat commented:
@bzbarsky-apple 
1. Yes once the fix is merged ill update the PR and add them back to the js files.
2. The commands and attribute where not supported when the scripts where added, I think we will rerun the tests and updates if they still fail for us, if they work fine we will update the scripts to enable automation .
3. Ill re-verify the test plan and the verification spread sheet, we will updated this if we had a outdated version.
4. Ok will make an issues.

bzbarsky-apple commented:
> Yes once the fix is merged ill update the PR and add them back to the js files.

Note that the fix is merged.

kowsisoundhar12 commented:
Added above scripts in separate PR and updated all the changes as per above suggestions

https://github.com/project-chip/connectedhomeip/pull/17513
https://github.com/project-chip/connectedhomeip/pull/17388
